### ISSUE TITLE
Inconsistency between :pwd and getcwd()

### ISSUE THREAD

**Issue by lervag** [neovim/neovim#9892]
### System information

- `nvim --version`: 0.3.4
- Vim (version: 8.1.1073) behaves differently? ~~Yes~~ **No**.
- Operating system/version: Arch Linux
- Terminal name/version: urxvt
- `$TERM`: xterm

### Steps to reproduce

Given `test.vim`:

```vim
function! Callback(id) abort
  call mkdir('subdir', 'p')

  silent edit $HOME/test/subdir/newfile
  set bufhidden=wipe
  silent lcd /home/lervag/test
  silent lcd -
  bnext

  echom printf('%-30s %-30s', getcwd(), execute('pwd')[1:])

  call delete('subdir', 'd')
endfunction

set autochdir

let s:test = []
silent edit $HOME/test/initial

call timer_start(200, 'Callback')
```

The output of `nvim -u test.vim` is

```
Actual:   /path/test/subdir       /path/test
Expected: /path/test              /path/test
```

See also: #9042, #7662

**Comment by mhinz** [comment]
Do you really have a different result with Vim? I used latest master and the version you specified and got the same result (`/path/test/subdir       /path/test`) with `nvim -u NORC -S test.vim` and `vim -u NORC -S test.vim`.

**Comment by lervag** [comment]
Sorry, my bad. I tested and thought I observed a different behaviour, but you are right: Vim and neovim behaves the same.

**Comment by blueyed** [comment]
Thansk for attacking this!
Does the assertion from https://github.com/neovim/neovim/issues/7662 fail here?

**Comment by mhinz** [comment]
If you mean `t-pwd-with-autocmd.vim`, then no. But it doesn't seem to fail with master either.

**Comment by mhinz** [comment]
Please check if the latest master works as expected for you!

**Comment by blueyed** [comment]
(Yeah, the test from #7662 does not work anymore, since the changing of the window in the FileType autocommand is not allowed anymore)

Thanks for the fix, but it seems to be related to autochdir only (which I am not using).

**Comment by justinmk** [comment]
@blueyed does that mean https://github.com/neovim/neovim/pull/9042 can be closed?

**Comment by blueyed** [comment]
@justinmk 
Well, since I am not using autochdir I think this fix here does not help me, but I've not noticed a problem with it for a while (but was not paying too much attention I guess).  Since #9042 does not really help in its current state I've closed it.

**Comment by lervag** [comment]
It seems the recent commit fixes the issue. Thanks! Looking forward to seeing 0.4.0 being released. Thank you for all the great work on maintaining and developing neovim!

**Comment by justinmk** [comment]
> Looking forward to seeing 0.4.0 being released

0.4.0 might take some time, but this fix will be in the next 0.3.x release. f891131c6bb2
